### PR TITLE
Rust Foundation Update: August 2024

### PR DESCRIPTION
**Pull request by JoelMarcey**

### CODE REVIEW

File: alpha/engagements/2024/Rust Foundation/update-2024-08.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+### Crate repo verification
+
+[Crate provenance](https://lawngno.me/blog/2024/06/10/divine-provenance.html) tracking, verifying that a given crate is actually associated with the repository it claims to be, is now running across the *entirety of the crates.io corpus*.
+
+The engineering to get this working across all of crates.io has surfaced various issues in underlying Rust libraries, particularly around threading, but these are being worked out and results of the entire corpus work should be available soon.
+
+### Deleting crates approved
+
+The crate deletion [RFC](https://github.com/rust-lang/rfcs/pull/3660) has been approved and merged. Implementation of the RFC details are now beginning. Recall, the purpose of the RFC is to provide crate owners with a mechanism to delete crates from crates.io under certain [conditions](https://github.com/Turbo87/rust-rfcs/blob/crates-io/crate-deletions/text/3660-crates-io-crate-deletions.md#proposal).
```
Comment (MarcoIeni) [inline alpha/engagements/2024/Rust Foundation/update-2024-08.md:13]: the pr isn't merged yet

**Review by MarcoIeni — APPROVED** [review]

File: alpha/engagements/2024/Rust Foundation/update-2024-08.md (hunk 1 above)
Comment (Turbo87) [reply alpha/engagements/2024/Rust Foundation/update-2024-08.md:13]: or is it... 😉

File: alpha/engagements/2024/Rust Foundation/update-2024-08.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+An effort is spinning up to examine how the Rust ecosystem uses build.rs in practice.
+
+Rust build.rs scripts are used for many different tasks — many of which require duplicated effort between crates, and all of which take place in an unsandboxed environment with full access to the build host. This effort is investigating whether a safer framework can be built that will allow build.rs scripts to be replaced by a unified framework, meaning that build.rs scripts become more standardized in practice and easier to flag for review in the same manner as many organizations currently review all `unsafe` blocks in their dependency graphs.
+
+## Announcements and Community
+
+### C++/Rust Interop Progress
+
+The initial C++/Rust Interop problem statement and high-level strategy has been written and has completed an internal review. There are three (3) high-level, concurrent strategies defined, which will serve as a basis for recruiting interested parties for both input and participation. We are striving for a top-down problem-space approach to drive strategic goals into tactical work
```
Comment (baumanj) [inline alpha/engagements/2024/Rust Foundation/update-2024-08.md:34]:
```suggestion
The initial C++/Rust Interop problem statement and high-level strategy has been written and has completed an internal review. There are three (3) high-level, concurrent strategies defined, which will serve as a basis for recruiting interested parties for both input and participation. We are striving for a top-down problem-space approach to drive strategic goals into tactical work.
```

Just a missing period
`</pedantry>`

**Review by baumanj — APPROVED** [review]
One minuscule typo noted

File: alpha/engagements/2024/Rust Foundation/update-2024-08.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+Crate provenance tracking continues with repo verification now being run across the entirety of the crates.io corpus, and a mechanism to delete crates has been approved and is now being implemented.
+
+## Engineering
+
+### Crate repo verification
+
+[Crate provenance](https://lawngno.me/blog/2024/06/10/divine-provenance.html) tracking, verifying that a given crate is actually associated with the repository it claims to be, is now running across the *entirety of the crates.io corpus*.
+
+The engineering to get this working across all of crates.io has surfaced various issues in underlying Rust libraries, particularly around threading, but these are being worked out and results of the entire corpus work should be available soon.
```
Comment (LawnGnome) [inline alpha/engagements/2024/Rust Foundation/update-2024-08.md:9]: I ended up blaming the wrong part of the stack, so:

```suggestion
The engineering to get this working across all of crates.io has surfaced various issues in underlying Rust libraries, most notably `gitoxide`, but these are being worked out and results of the entire corpus work should be available soon.
```

**Pull request closed without merging**